plugins/backup/veeam/src/main/java/org/apache/cloudstack/backup/veeam/VeeamClient.java

```diff
@@ -0,0 +1,661 @@
+            if (jobObjects == null || jobObjects.getObjects() == null) {
+                LOG.warn("No objects found in the Veeam job " + jobId);
+                return false;
+            }
+            for (final ObjectInJob jobObject : jobObjects.getObjects()) {
+                if (jobObject.getName().equals(vmwareInstanceName) && jobObject.getHierarchyObjRef().equals(veeamVmRefId)) {
+                    final HttpResponse deleteResponse = delete(String.format("/jobs/%s/includes/%s", jobId, jobObject.getObjectInJobId()));
+                    return checkTaskStatus(deleteResponse);
+                }
+            }
+            LOG.warn(vmwareInstanceName + " VM was not found to be attached to Veaam job (backup offering): " + jobId);
+            return false;
+        } catch (final IOException e) {
+            LOG.error("Failed to list Veeam jobs due to:", e);
+            checkResponseTimeOut(e);
+        }
+        return false;
+    }
+
+    public boolean restoreFullVM(final String vmwareInstanceName, final String restorePointId) {
+        LOG.debug("Trying to restore full VM: " + vmwareInstanceName + " from backup");
+        try {
+            final HttpResponse response = post(String.format("/vmRestorePoints/%s?action=restore", restorePointId), null);
+            return checkTaskStatus(response);
+        } catch (final IOException e) {
+            LOG.error("Failed to restore full VM due to: ", e);
+            checkResponseTimeOut(e);
+        }
+        throw new CloudRuntimeException("Failed to restore full VM from backup");
+    }
+
+    /////////////////////////////////////////////////////////////////
+    //////////////// Public Veeam PS based APIs /////////////////////
+    /////////////////////////////////////////////////////////////////
+
+    /**
+     * Generate a single command to be passed through SSH
+     */
+    protected String transformPowerShellCommandList(List<String> cmds) {
+        StringJoiner joiner = new StringJoiner(";");
+        joiner.add("PowerShell Add-PSSnapin VeeamPSSnapin");
+        for (String cmd : cmds) {
+            joiner.add(cmd);
+        }
+        return joiner.toString();
+    }
+
+    /**
+     * Execute a list of commands in a single call on PowerShell through SSH
+     */
+    protected Pair<Boolean, String> executePowerShellCommands(List<String> cmds) {
+        try {
+            // FIXME: make ssh timeouts configurable
+            Pair<Boolean, String> pairResult = SshHelper.sshExecute(veeamServerIp, veeamServerPort,
+                    veeamServerUsername, null, veeamServerPassword,
+                    transformPowerShellCommandList(cmds),
+                    120000, 120000, 900000);
+            return pairResult;
+        } catch (Exception e) {
+            throw new CloudRuntimeException("Error while executing PowerShell commands due to: " + e.getMessage());
+        }
+    }
+
+    public boolean setJobSchedule(final String jobName) {
```

Comment:
This method was confusing to me, found it is only used on unit tests. Can it be renamed?

Reply:
Removed